articles/implementing-firebase-in-flutter/index.md

```diff
@@ -0,0 +1,278 @@
+
+Download the generated `google-services.json` file and paste it into the `app` folder
+
+![paste json](/engineering-educationimplementing-firebase-in-flutter/paste-json.png)
+
+Paste `classpath 'com.google.gms:google-services:4.3.3'` in the project level – `build.gradle` file.
+
+```
+dependencies {
+  classpath 'com.android.tools.build:gradle:3.5.0'
+  classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+  classpath 'com.google.gms:google-services:4.3.3'
+}
+```
+
+Navigate to the app-level – `build.gradle` file and add `apply plugin: 'com.google.gms.google-services'` as shown below.
+
+```
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
+apply plugin: 'com.google.gms.google-services'
+````
+
+After the project is created successfully, navigate back to the Firebase console's realtime database section, and create a new database.
+
+![create database](/engineering-educationimplementing-firebase-in-flutter/create-database.png)
+
+In the `rules` section, ensure that you click on `test mode` rather than the `locked mode`. This allows us to read and write data without authentication. Note that these rules should not be used in a production application since anyone can access data.
+
+```json
+{
+  "rules": {
+    ".read": "now < 1610053200000",  // 2021-1-8
+    ".write": "now < 1610053200000",  // 2021-1-8
+  }
+}
+```
+
+We have successfully set up Firebase and the flutter project. Let's design the UI.
+
+### Creating UI
+The app will have a simple user interface. It will allow a user to input a word or sentence and click on a button to save it to the Firebase database.
+
+![ui](/engineering-educationimplementing-firebase-in-flutter/ui.jpeg)
+
+Create a new file called `home.dart.` In this file, create a `stateful class` and name it `HomePage.`
+We create a stateful widget and scaffold, as shown below.
+
+```dart 
+class MyApp extends StatefulWidget { //creating statefulwidget 
+  MyApp({Key key}) : super(key: key);
+
+  @override
+  _MyAppState createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold();  // add scaffold here
+  }
+}
+```
+
+We can then create the above `UI` in this file by pasting the following code in the body section of the `Scaffold.` 
+
+```dart
+Container(
+  child: Column(
+    children: <Widget>[
+      SizedBox(height: 250.0),
+      Padding(
+        padding: EdgeInsets.all(10.0),
+        child: TextField(
+        ),
+      ),
+      SizedBox(height: 30.0),
+      Center(
+          child: RaisedButton(
+              color: Colors.pinkAccent,
+              child: Text("Save to Database"),
+              onPressed: () {
+                addData(textcontroller.text);
+                //call method flutter upload
+              })),
```

Comment:
Instruct the users to add the `addData` function in the `onPressed` when you reach the store data section.
If the reader runs the app at this point, it will throw an error.